src/Util/ProjectUpdateTrait.php

```diff
@@ -0,0 +1,154 @@
+{
+
+    /**
+     * Update project info (codeowners and support level badge).
+     */
+    protected function updateProjectInfo($api, $project, $baseBranch, $branchName, $commitMessage, $prTitle, $prBody, $logger, $supportLevelBadge = '', $codeowners = '')
+    {
+        $branchToClone = $baseBranch;
+        if (count(explode('/', $project)) != 2) {
+            throw new \Exception("Invalid project name: $project");
+        }
+        if (empty($codeowners) && empty($supportLevelBadge)) {
+            throw new \Exception("Must specify at least one of codeowners or support-level-badge");
+        }
+        $url = "git@github.com:$project.git";
+        $remote = new Remote($url);
+        $dir = sys_get_temp_dir() . '/update-tool/' . $remote->project();
+
+        $existingPrFound = false;
+
+        $prs = $api->matchingPRs($project, $prTitle)->prNumbers();
+        if (count($prs) === 1) {
+            $prNumber = reset($prs);
+            $parts = explode('/', $project);
+            $fullPr = $api->prGet($parts[0], $parts[1], $prNumber);
+            if ($fullPr) {
+                $branchToClone = $fullPr['head']['ref'];
+                $branchName = $branchToClone;
+                $existingPrFound = true;
+            }
+        }
+
+
+        $workingCopy = WorkingCopy::cloneBranch($url, $dir, $branchToClone, $api);
+
+        $workingCopy->setLogger($logger);
+
+        if (!$existingPrFound) {
+            $workingCopy->createBranch($branchName);
+            $workingCopy->switchBranch($branchName);
+        }
+        $codeowners_changed = false;
+        $readme_changed = false;
+
+        if (!empty($codeowners)) {
+            // Append given CODEOWNERS line.
+            $string_to_add = '* ' . $codeowners . "\n";
+            $codeowners_content = '';
+            if (file_exists("$dir/CODEOWNERS")) {
+                $codeowners_content = file_get_contents("$dir/CODEOWNERS");
+            }
+            if (strpos($codeowners_content, $string_to_add) === false) {
+                file_put_contents("$dir/CODEOWNERS", $string_to_add, FILE_APPEND);
+                $workingCopy->add("$dir/CODEOWNERS");
+                $codeowners_changed = true;
+            }
+        }
+
+        if (!empty($supportLevelBadge)) {
+            $badge_contents = SupportLevel::getSupportLevelBadge($supportLevelBadge);
+            if (!$badge_contents) {
+                throw new \Exception("Invalid support level badge: $supportLevelBadge.");
+            }
+            if (file_exists("$dir/README.md")) {
+                $readme_contents = file_get_contents("$dir/README.md");
+            } else {
+                $readme_contents = '';
+            }
+
+            if (!SupportLevel::compareSupportLevelFromReadmeAndBadge($readme_contents, $badge_contents)) {
+                $lines = explode("\n", $readme_contents);
+                [$badge_insert_line, $empty_line_after] = $this->getBadgeInsertLine($lines, $badge_contents);
```

Comment:
Is this idempotent? It sort of looks like you'll end up with two support level badges if you run this on a README that already has a support level badge.

Reply:
(At a different support level, that is)

Reply:
That wouldn't run because of SupportLevel::compareSupportLevelFromReadmeAndBadge

Reply:
ohhh, different support level. What should be the outcome with different levels?

Reply:
Replace the existing support level badge with a new one.

Reply:
Fixed